esrally/racecontrol.py

```diff
@@ -386,9 +387,14 @@ def run(cfg):
     logger = logging.getLogger(__name__)
     name = cfg.opts("race", "pipeline")
 
-    if console.RALLY_RUNNING_IN_DOCKER:
+    if os.environ.get("RALLY_RUNNING_IN_DOCKER", "").upper() == "TRUE":
         # in this case only benchmarking remote Elasticsearch clusters makes sense
-        cfg.add(config.Scope.applicationOverride, "race", "pipeline", "benchmark-only")
+        if name != "benchmark-only":
+            raise exceptions.SystemSetupError(
+                "Only the [benchmark-only] pipeline is supported by the Rally Docker image.\n"
+                "Add --pipeline=benchmark-only in your Rally arguments and try again.\n"
+                "For more details read the docs for the benchmark-only pipeline in {}\n".format(
+                    urllib.parse.urljoin(DOC_LINK, "pipelines.html#benchmark-only")))
```

Comment:
The nice property of `urljoin` is that it manages trailing slashes, as required, by itself.
With the standard way we've been doing things, i.e. string join, the resulting URL has two double forward slashes (due to https://github.com/elastic/rally/blob/646cc1e0f3edd06742e027461e90459b0a56005f/esrally/__init__.py#L29-L31).

For example the resulting URL in https://github.com/elastic/rally/blob/646cc1e0f3edd06742e027461e90459b0a56005f/esrally/racecontrol.py#L405 produces something like `...//pipelines.html#from-distribution`

Reply:
I am not 100% convinced we need this but instead we should fix the formatting of the other link as we have full control of the prefix? In any case I think we should either offer a consistent helper function across the code base (I have no good answer where this would belong; `console`, `convert` and `io` are possible candidates but none of them is a good fit IMHO) or refrain from providing doc links in exception messages?

Reply:
As discussed we'll tackle the rest of the places where this needs to be adjusted in a separate PR.